tlv/tlv_type_param.go

```diff
@@ -0,0 +1,16 @@
+package tlv
```

Comment:
Yeah I think this is right. At least up to uint16. I think maybe though that there is a limit to the number of TLV records you can attach to a message due to size requirements, but my rough estimation of that is 10k.

Reply:
> It ends up using go generate to achieve this. We won't need to run it again, as due to implementation limits, we can only have 100 members in a union struct. If we need more, we can break it up (I think?).

The limitation looks like it comes from go, not from the protocol. I think this is probably fine in practice for us though. Since TLV types are message specific, to blow through this we would need to have  >50 optional/required types.

Reply:
They are, but have we ever used anything beyond like _10_ so far? Granted this is a limitation with this approach, but I think we this, we can get rid of all the existing type wrappers around primitive types just. The union is limited to 100 value, but I think you can union two instances that have a 100 values each? 

This covers all the "official" TLV values, but not the experimental range. Today I don't think we have any TLV values in _wire_ messages that have eaten into the experimental range. 

> Since TLV types are message specific, to blow through this we would need to have >50 optional/required types.

The message specific name spacing can be expressed with either a type alias, or just implicitly as you define the type int+value within the struct itself.